src/components/Publish/_validation.ts

```diff
@@ -54,25 +56,56 @@ const validationService = {
   })
 }
 
+const maxDecimalsValidation = new RegExp(
+  '^\\d+(\\.\\d{1,' + MAX_DECIMALS + '})?$'
+)
+const swapFeeMaxDecimalsValidation = new RegExp(
+  '^\\d+(\\.\\d{1,' + (MAX_DECIMALS - 1) + '})?$'
+)
+
 const validationPricing = {
   type: Yup.string()
     .matches(/fixed|dynamic|free/g, { excludeEmptyString: true })
     .required('Required'),
   // https://github.com/jquense/yup#mixedwhenkeys-string--arraystring-builder-object--value-schema-schema-schema
+
   price: Yup.number()
     .min(1, (param: { min: number }) => `Must be more or equal to ${param.min}`)
+    .max(
+      1000000,
+      (param: { max: number }) => `Must be less than or equal to ${param.max}`
+    )
+    .test(
+      'maxDigitsAfterDecimal',
+      `Must have maximum ${MAX_DECIMALS} decimal digits`,
+      (param) => maxDecimalsValidation.test(param?.toString())
+    )
     .required('Required'),
   amountDataToken: Yup.number()
     .min(50, (param) => `Must be more or equal to ${param.min}`)
     .required('Required'),
   amountOcean: Yup.number()
     .min(50, (param) => `Must be more or equal to ${param.min}`)
+    .max(
+      1000000,
+      (param: { max: number }) => `Must be less than or equal to ${param.max}`
+    )
+    .test(
+      'maxDigitsAfterDecimal',
+      `Must have maximum ${MAX_DECIMALS} decimal digits`,
+      (param) => maxDecimalsValidation.test(param?.toString())
+    )
     .required('Required'),
   weightOnDataToken: Yup.string().required('Required'),
   weightOnOcean: Yup.string().required('Required'),
   swapFee: Yup.number()
     .min(0.1, (param) => `Must be more or equal to ${param.min}`)
     .max(10, 'Maximum is 10%')
+    .test(
+      'maxDigitsAfterDecimal',
+      `Must have maximum ${MAX_DECIMALS - 1} decimal digits`,
```

Comment:
Got an  error on publish for 5 decimals on swap fee. It worked fine with 4 decimals

Reply:
`MAX_DECIMALS`  is 5 so, 5-1 =4  . i think  `MAX_DECIMALS`   should be 6, no idea why i set it to 5

Reply:
should i change it in the constants.ts file?

Reply:
yes please